driver-core/src/test/functional/com/mongodb/ClusterFixture.java

```diff
@@ -396,32 +400,41 @@ public static StreamFactory getStreamFactory() {
     }
 
     public static StreamFactory getAsyncStreamFactory() {
-        String streamType = System.getProperty("org.mongodb.test.async.type", "nio2");
-
-        if (streamType.equals("netty")) {
-            return new NettyStreamFactory(getSocketSettings(), getSslSettings());
-        } else if (streamType.equals("nio2")) {
+        StreamFactoryFactory overriddenStreamMetafactory = getOverriddenStreamFactoryFactory();
+        if (overriddenStreamMetafactory == null) { // use NIO2
             if (getSslSettings().isEnabled()) {
                 return new TlsChannelStreamFactoryFactory().create(getSocketSettings(), getSslSettings());
             } else {
                 return new AsynchronousSocketChannelStreamFactory(getSocketSettings(), getSslSettings());
             }
         } else {
-            throw new IllegalArgumentException("Unsupported stream type " + streamType);
+            return assertNotNull(overriddenStreamMetafactory).create(getSocketSettings(), getSslSettings());
         }
     }
 
     @Nullable
     public static StreamFactoryFactory getOverriddenStreamFactoryFactory() {
         String streamType = System.getProperty("org.mongodb.test.async.type", "nio2");
 
-        if (streamType.equals("netty")) {
-            if (nettyStreamFactoryFactory == null) {
-                nettyStreamFactoryFactory = NettyStreamFactoryFactory.builder().build();
+        if (nettyStreamFactoryFactory == null) {
+            if (streamType.equals("netty")) {
```

Comment:
this double if statement could be joined.

Reply:
Thanks! Done.